packages/logging/src/logging-and-telemetry.ts

```diff
@@ -114,10 +155,24 @@ export class MongoshLoggingAndTelemetry {
     this.trackingProperties.session_id = this.log.logId;
 
     this.isDummyLog = false;
+    if (this.hookedExternalListenersLogId !== this.log.logId) {
+      hookLogger(
+        {
+          ...this.bus,
```

Comment:
this is quite hacky... we probably should rather update the `hookLogger` instead?

i.e. instead of `this.log` we could update it to take `() => this.log` which can also support nullability

Reply:
alternatively we call `this.bus.removeAllEventListeners()` when stopping logging and set up the event listeners again for the rest

Reply:
oh yeah, didn't think about having the writer be the adapter, sounds good